features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/RumFeature.kt

```diff
@@ -399,6 +425,24 @@ internal class RumFeature(
         }
     }
 
+    private fun resolveUploadBlockerProperties(event: UploadQualityEvent): Map<String, Any?> {
+        val properties = mutableMapOf<String, Any?> (
+            KEY_METRIC_TYPE to UploadBlockerMetric.BATCH_BLOCKED_KEY.key,
+            UploadBlockerMetric.TRACK_KEY.key to event.track,
+            UploadBlockerMetric.UPLOAD_DELAY_KEY.key to event.uploadDelay
+        )
+
+        if (event is UploadQualityEvent.UploadQualityBlockerEvent) {
+            properties[UploadBlockerMetric.BATCH_COUNT_KEY.key] = event.batchCount
+            properties[UploadBlockerMetric.BLOCKERS_KEY.key] = event.blockers
+        } else if (event is UploadQualityEvent.UploadQualityFailureEvent) {
+            properties[UploadBlockerMetric.BATCH_COUNT_KEY.key] = event.batchCount
+            properties[UploadBlockerMetric.FAILURE_KEY.key] = event.failure
+        }
+
+        return properties.toMap()
```

Comment:
why an additional call to `toMap` is needed? It seems to me that it is redundant.